common/utils/src/main/java/org/graalvm/buildtools/utils/NativeImageUtils.java

```diff
@@ -102,4 +109,44 @@ public static String escapeArg(String arg) {
         }
         return arg;
     }
+
+    /**
+     *
+     * @param requiredVersion Required version can be {@code MAJOR}, {@code MAJOR.MINOR} or {@code MAJOR.MINOR.PATCH}
+     * @param versionToCheck The version to check, as returned by {@code native-image --version}
+     * @throws IllegalStateException when the version is not correct
+     */
+    public static void checkVersion(String requiredVersion, String versionToCheck) {
+        Matcher requiredMatcher = requiredVersionPattern.matcher(requiredVersion);
+        if (!requiredMatcher.matches()) {
+            throw new IllegalArgumentException("Invalid version " + requiredVersion + ", should be for example \"22\", \"22.3\" or \"22.3.0\".");
+        }
+        Matcher checkedMatcher = graalvmVersionPattern.matcher(versionToCheck);
+        if (!checkedMatcher.matches()) {
+            throw new IllegalArgumentException("Version to check " + versionToCheck + " can't be parsed.");
+        }
+        int requiredMajor = Integer.parseInt(requiredMatcher.group(1));
```

Comment:
:o: This could be refactored to have a method which checks a single number, then call it for major, minor and patch.

Reply:
Given the custom message in the logs (which is IMO important), I am not sure, could you please double check?

Reply:
Yeah, we could use a static method with lambdas, but I'm not sure it would be easier to read.